### PR TITLE
OCLOMRS-536: Previewing a Concept WIth Both Sets and Answers Does not Show Sets

### PR DESCRIPTION
**Pull request by myrdstom**
# JIRA TICKET NAME:
[Previewing a Concept WIth Both Sets and Answers Does not Show Answers](https://issues.openmrs.org/browse/OCLOMRS-536)

# Summary:
Previewing a concept currently assumes that only an Answer or Sets can be present at a time. This PR caters for the possibility of both Answers and sets being available and displays them both.

### CODE REVIEW

File: src/components/bulkConcepts/component/PreviewCard.jsx [hunk 1]
```diff
@@ -82,6 +82,23 @@ const PreviewCard = ({
         remainingItems={answers.slice(NUMBER_OF_ITEMS_TO_DISPLAY).join(' ')}
       />;
     }
+    if (answers && answers.length > 0 && setMembers && setMembers.length > 0) {
```
Comment (dkayiwa) [inline src/components/bulkConcepts/component/PreviewCard.jsx]: What happens when answers.length > 0 but setMembers.length = 0?

Comment (myrdstom) [reply src/components/bulkConcepts/component/PreviewCard.jsx]: Line 76 handles this scenario. But ideally what would happen is only answer mappings would be displayed

Comment (dkayiwa) [reply src/components/bulkConcepts/component/PreviewCard.jsx]: My question is, if answers.length > 0 but setMembers.length = 0, does this mean you would not render answers?

Comment (myrdstom) [reply src/components/bulkConcepts/component/PreviewCard.jsx]: `if (answers && answers.length > 0 && setMembers && setMembers.length < 1)`
This condition would pass in this scenario, and hence the answers would be displayed. Would it be more satisfactory if I equated the code to something like
(answers && answers.length > 0 && setMembers && setMembers.length = 0?
 My understanding is it is similar to the code I currently have. Or am I missing something else?

Comment (dkayiwa) [reply src/components/bulkConcepts/component/PreviewCard.jsx]: When you say that the condition would pass, which line are you talking about?

Comment (myrdstom) [reply src/components/bulkConcepts/component/PreviewCard.jsx]: Im talking about `line 76` in file `src/components/bulkConcepts/component/PreviewCard.jsx`

Comment (dkayiwa) [reply src/components/bulkConcepts/component/PreviewCard.jsx]: My questions are in reference to line 85

Comment (myrdstom) [reply src/components/bulkConcepts/component/PreviewCard.jsx]: I have refactored the code to use `if else-if` statements. Would that cater for your concern?

Comment (dkayiwa) [reply src/components/bulkConcepts/component/PreviewCard.jsx]: What do your team mates say?

Comment (myrdstom) [reply src/components/bulkConcepts/component/PreviewCard.jsx]: I had afew suggestions, is it possible to give it a final glance?

Comment (Karuhanga) [reply src/components/bulkConcepts/component/PreviewCard.jsx]: > My question is, if answers.length > 0 but setMembers.length = 0, does this mean you would not render answers?



> `if (answers && answers.length > 0 && setMembers && setMembers.length < 1)`
> This condition would pass in this scenario, and hence the answers would be displayed. Would it be more satisfactory if I equated the code to something like
> answers && answers.length > 0 && setMembers && setMembers.length = 0?
> My understanding is it is similar to the code I currently have. Or am I missing something else?

I believe this situation is catered for in the scenario Paul outlines, @dkayiwa. A little clarity on what we might be missing perhaps?

Comment (dkayiwa) [reply src/components/bulkConcepts/component/PreviewCard.jsx]: Why do you combine the check for setmembers and answers in the same if condition?

Comment (myrdstom) [reply src/components/bulkConcepts/component/PreviewCard.jsx]: Thank you for the clarity, I should effect the changes shortly

Comment (myrdstom) [reply src/components/bulkConcepts/component/PreviewCard.jsx]: I have refactored the code please have a look at it when you get some time

**Pull request merged**